org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateResourceService.mtl

```diff
@@ -298,35 +299,27 @@ import [javaClassFullName(aResource, anAdaptorInterface, null) /];
-            // [protected (dialogMethodName(aSelectionDialog, true).concat('_init'))]
-            // [/protected]
+        // [protected (dialogMethodName(aSelectionDialog, true).concat('_init'))]
+        // [/protected]
 
-            httpServletRequest.setAttribute("selectionUri",UriBuilder.fromUri(OSLC4JUtils.getServletURI()).path(uriInfo.getPath()).build().toString());
-            // [protected (dialogMethodName(aSelectionDialog, true).concat('_setAttributes'))]
-            // [/protected]
+        httpServletRequest.setAttribute("selectionUri",UriBuilder.fromUri(OSLC4JUtils.getServletURI()).path(uriInfo.getPath()).build().toString());
+        // [protected (dialogMethodName(aSelectionDialog, true).concat('_setAttributes'))]
+        // [/protected]
 
-            if (terms != null ) {
-                httpServletRequest.setAttribute("terms", terms);
-                final List<[dialogMethodResourceType(aSelectionDialog)/]> resources = [javaClassNameForAdaptorManager(anAdaptorInterface) /].[dialogMethodName(aSelectionDialog, true)/](httpServletRequest[commaSeparate(dialogMethodParameterList(aSelectionDialog, true), true, false)/], terms);
-                if (resources!= null) {
-                            httpServletRequest.setAttribute("resources", resources);
-                            RequestDispatcher rd = httpServletRequest.getRequestDispatcher("[resourceSelectorResultsJspRelativeFileName(aSelectionDialog) /]");
-                            rd.forward(httpServletRequest, httpServletResponse);
-                }
-                //a empty search should return an empty list and not NULL!
-                throw new WebApplicationException(Status.NOT_FOUND);
-
-            } else {
-                try {
-                    RequestDispatcher rd = httpServletRequest.getRequestDispatcher("[resourceSelectorJspRelativeFileName(aSelectionDialog) /]");
-                    rd.forward(httpServletRequest, httpServletResponse);
-                } catch (Exception e) {
-                    throw new ServletException(e);
-                }
+        if (terms != null ) {
+            httpServletRequest.setAttribute("terms", terms);
+            final List<[dialogMethodResourceType(aSelectionDialog)/]> resources = [javaClassNameForAdaptorManager(anAdaptorInterface) /].[dialogMethodName(aSelectionDialog, true)/](httpServletRequest[commaSeparate(dialogMethodParameterList(aSelectionDialog, true), true, false)/], terms);
+            if (resources!= null) {
+                        httpServletRequest.setAttribute("resources", resources);
+                        RequestDispatcher rd = httpServletRequest.getRequestDispatcher("[resourceSelectorResultsJspRelativeFileName(aSelectionDialog) /]");
+                        rd.forward(httpServletRequest, httpServletResponse);
             }
-        } catch (Exception e) {
-            throw new WebApplicationException(e);
+            //a empty search should return an empty list and not NULL!
```

Comment:
Isn't it a 500?

Reply:
should also be logged:

    log.error("A search should return an empty List and not 'null' if nothing was found!")

Reply:
Please see new commit.